math/calculator.cpp

```diff
@@ -0,0 +1,232 @@
+                std::cout << "Type a number: ";
+                std::cin >> number;
+
+                /**
+                 * Set the first number as the main number.
+                 * Save it in the new node so it's not re-writed.
+                 */
+
+                if (i == 1) {
+                    new_node->datatype = number;
+
+                /**
+                 * If the current number is not the first one,
+                 * subtract the current one with the previous one.
+                 */
+
+                } else {
+                    new_node->datatype -= number;
+                }
+            }
+
+            std::cout << "\nThe subtraction of the numbers is: "
+                << new_node->datatype << std::endl;
+
+            pause();
+            break;
+
+        case 3:
+            std::cout << "Type the amount of numbers to multiply: ";
+            std::cin >> amount_to_multiply;
+
+            std::cout << "\n";
+
+            for (int i = 1; i <= amount_to_multiply; i++) {
+                std::cout << "Type a number: ";
+                std::cin >> number;
+
+                /**
+                 * Set the first number as the main number.
+                 * Save it in the new node so it's not re-writed.
+                 */
+
+                if (i == 1) {
+                    new_node->datatype = number;
+
+                /**
+                 * If the current number is not the first one,
+                 * multiply the current one with the previous one.
+                 */
+
+                } else {
+                    new_node->datatype *= number;
+                }
+            }
+
+            std::cout << "\nThe multiply result of the numbers is: "
+                << new_node->datatype << std::endl;
+
+            pause();
+            break;
+
+        case 4:
+            std::cout << "Type the amount of numbers to divide: ";
+            std::cin >> amount_to_divide;
+
+            std::cout << "\n";
+
+            for (int i = 1; i <= amount_to_divide; i++) {
+                std::cout << "Type a number: ";
+                std::cin >> number;
+
+                /**
+                 * Set the first number as the main number.
+                 * Save it in the new node so it's not re-writed.
+                 */
+
+                if (i == 1) {
+                    new_node->datatype = number;
+
+                /**
+                 * If the current number is not the first one,
+                 * divide the previous number with the current one.
+                 */
+
+                } else {
+                    new_node->datatype /= number;
+                }
+            }
+
+            std::cout << "\nThe division of the numbers is: "
+                << new_node->datatype << std::endl;
+
+            pause();
+            break;
+        }
+
+        /**
+         * Check user's OS
+         * Use system("clear") if OS is GNU/Linux based
+         */
+
+        #ifdef __linux__
```

Comment:
lines 221-230 can be eliminated. They simply do not add any value to the code and the problem.

Reply:
Why shall that be removed? Output would look wrong.

Reply:
there would be no change in the output. This command is only meant to clear the terminal screen. If removed, the results will scroll. 

Please note that this repository is meant for code with algorithms meant for education and not for creating UI elements.